initializr-generator/src/main/java/io/spring/initializr/generator/buildsystem/maven/MavenBuild.java

```diff
@@ -134,4 +138,110 @@ public String getPackaging() {
 		return this.packaging;
 	}
 
+	public void resource(String directory, Consumer<ResourceBuilder> customizer) {
+		customizer.accept(this.resources.computeIfAbsent(directory, (key) -> new ResourceBuilder(directory)));
+	}
+
+	public List<Resource> getResources() {
+		return this.resources.values().stream().map(ResourceBuilder::build).collect(Collectors.toList());
+	}
+
+	/**
+	 *
+	 * Builder to create a {@link Resource}.
+	 *
+	 *
+	 */
+	public static final class ResourceBuilder {
+
+		private String directory;
+
+		private String targetPath;
+
+		private Boolean filtering;
+
+		private List<String> includes = new ArrayList<>();
+
+		private List<String> excludes = new ArrayList<>();
+
+		public ResourceBuilder(String directory) {
+			this.directory = directory;
+		}
+
+		Resource build() {
+			return new Resource(this.directory, this.targetPath, this.filtering, this.includes, this.excludes);
+		}
+
+		public ResourceBuilder include(List<String> includes) {
+			this.includes = includes;
+			return this;
+		}
+
+		public ResourceBuilder targetPath(String targetPath) {
+			this.targetPath = targetPath;
+			return this;
+		}
+
+		public ResourceBuilder filtering(Boolean filtering) {
```

Comment:
I don't think it should use the wrapper type. The default should match what Maven does (`false`).